src/cryptonote_core/miner.cpp

```diff
@@ -408,5 +528,313 @@ namespace cryptonote
+  uint8_t miner::get_idle_threshold() const
+  {
+    return m_idle_threshold;
+  }
+  //-----------------------------------------------------------------------------------------------------
+  bool miner::set_idle_threshold(const uint8_t& idle_threshold)
+  {
+    if(idle_threshold > BACKGROUND_MINING_MAX_IDLE_THRESHOLD_PERCENTAGE) return false;
+    if(idle_threshold < BACKGROUND_MINING_MIN_IDLE_THRESHOLD_PERCENTAGE) return false;
+    m_idle_threshold = idle_threshold;
+    return true;
+  }
+  //-----------------------------------------------------------------------------------------------------
+  uint8_t miner::get_mining_threshold() const
+  {
+    return m_mining_threshold;
+  }
+  //-----------------------------------------------------------------------------------------------------
+  bool miner::set_mining_threshold(const uint8_t& mining_threshold)
+  {
+    if(mining_threshold > BACKGROUND_MINING_MAX_MINING_THRESHOLD_PERCENTAGE) return false;
+    if(mining_threshold < BACKGROUND_MINING_MIN_MINING_THRESHOLD_PERCENTAGE) return false;
+    m_mining_threshold = mining_threshold;
+    return true;
+  }
+  //-----------------------------------------------------------------------------------------------------
+  bool miner::get_cpu_stats(cpu_data& cpud)
+  {
+    #if defined(WIN32) || defined(_WIN32) || defined(__WIN32)
+
+    LOG_PRINT_RED_L0("background mining not supported on windows ... yet");
+    return false;
+
+    #else
+
+    const std::string STR_CPU("cpu");
+    const std::size_t STR_CPU_LEN = STR_CPU.size();
+    const std::string STAT_FILE_PATH = "/proc/stat";
+    const std::string PROC_FILE_PATH = "/proc/self/stat";
+
+    if( !epee::file_io_utils::is_file_exist(STAT_FILE_PATH) )
+    {
+      LOG_PRINT_RED_L0(STAT_FILE_PATH << " file does not exist");
+      return false;
+    }
+
+    std::ifstream stat_file_stream(STAT_FILE_PATH);
+    if( stat_file_stream.fail() )
+    {
+      LOG_PRINT_RED_L0("failed to read " << STAT_FILE_PATH);
+      return false;
+    }
+
+    cpud.update_time = misc_utils::get_tick_count();
+    std::string line;
+    std::getline(stat_file_stream, line);
+    std::istringstream stat_file_iss(line);
+    std::string cpu_label;
+
+    stat_file_iss >> cpu_label;
+    for(size_t i = 0; i < CPU_DATA_OVERALL_TIMES_COUNT; ++i)
+    {
+      stat_file_iss >> cpud.overall_cpu_times[i];
+    }
+
+
+    std::ifstream proc_file_stream(PROC_FILE_PATH);
+    if( proc_file_stream.fail() )
+    {
+      LOG_PRINT_RED_L0("failed to read " << PROC_FILE_PATH);
+      return false;
+    }
+
+    std::getline(proc_file_stream, line);
+    std::istringstream proc_file_iss(line);
+    std::string tmp;
+    for(size_t i = 0; i < 13; ++i)
+    {
+      proc_file_iss >> tmp;
+      //LOG_PRINT_RED_L0(" ( " << i << ") --> " << tmp);
+      //proc_file_iss.ignore(65536, ' ');
+    }
+
+    float utime, stime;
+    proc_file_iss >> utime;
```

Comment:
These should be the same values returned by the posix system call `times`, which hopefully works across all `!Win32` systems.